src/cheri_insts.sail

```diff
@@ -1165,16 +1165,24 @@ function handle_store_cap_via_cap(rs, cs, cap_val, vaddrBits) = {
     RETIRE_FAIL
   } else match translateAddr(vaddrBits, Write(if rs_val.tag then Cap else Data)) {
     TR_Failure(e, _) => { handle_mem_exception(vaddrBits, e); RETIRE_FAIL },
-    TR_Address(addr, _) => {
-      let eares : MemoryOpResult(unit) = mem_write_ea_cap(addr, aq, rl, false);
-      match (eares) {
-        MemException(e) => { handle_mem_exception(addr, e); RETIRE_FAIL },
-        MemValue(_) => {
-          let res : MemoryOpResult(bool) = mem_write_cap(addr, rs_val, aq, rl, false);
-          match (res) {
-            MemValue(true)  => RETIRE_SUCCESS,
-            MemValue(false) => internal_error("store got false from mem_write_value"),
-            MemException(e) => { handle_mem_exception(addr, e); RETIRE_FAIL }
+    TR_Address(addr, caveats) => {
+      match (caveats.ptw_sc_mod, rs_val.tag) {
+        (PTW_SCM_TRAP, true) => {
+          handle_cheri_cap_exception(CapEx_TLBNoStoreCap, cs);
```

Comment:
`PTW_SCM_TRAP` can be hidden in the PTW logic, but until ReadWrite has its load/store sides split, so that we can indicate whether the store is tag-asserting or tag-clearing (rather than merely being tag-capable), we will erroneously trap on amoswap.c that store a cleared tag.